coriolis/tests/osmorphing/osdetect/test_windows.py

```diff
@@ -0,0 +1,207 @@
+        (WIN_VERSION_PS_OUTPUT_MISSING_MAJOR_VERSION,
+         "6.3.20348",
+         "ServerDatacenterEval",
+         "Server",
+         "Windows Server 2022 Datacenter Evaluation")
+    )
+    @ddt.unpack
+    @mock.patch.object(windows.WindowsOSDetectTools, '_load_registry_hive')
+    @mock.patch.object(windows.WindowsOSDetectTools, '_unload_registry_hive')
+    @mock.patch.object(windows.uuid, 'uuid4')
+    def test__get_image_version_info(
+            self, ps_output, version, product_name, product_type,
+            product_full_name, mock_uuid4, mock_unload_registry_hive,
+            mock_load_registry_hive):
+        self.conn.exec_ps_command.return_value = (
+            ps_output.replace('\n', os.linesep)
+        )
+
+        result = self.windows_os_detect_tools._get_image_version_info()
+
+        mock_load_registry_hive.assert_called_once_with(
+            "HKLM\\%s" % mock_uuid4.return_value,
+            "%sWindows\\System32\\config\\SOFTWARE" % self.os_root_dir)
+
+        mock_unload_registry_hive.assert_called_once_with(
+            "HKLM\\%s" % mock_uuid4.return_value)
+
+        expected_result = (
+            windows.version.LooseVersion(version),
+            product_name,
+            product_type,
+            product_full_name
+        )
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(windows.WindowsOSDetectTools, '_load_registry_hive')
+    @mock.patch.object(windows.WindowsOSDetectTools, '_unload_registry_hive')
+    def test__get_image_version_info_with_exception(
+            self, mock_unload_registry_hive,
+            mock_load_registry_hive):
+        self.conn.exec_ps_command.return_value = (
+            WIN_VERSION_PS_OUTPUT_MISSING_FIELDS.replace('\n', os.linesep))
+
+        mock_unload_registry_hive.assert_not_called()
+        mock_load_registry_hive.assert_not_called()
+
+        self.assertRaises(
+            exception.CoriolisException,
+            self.windows_os_detect_tools._get_image_version_info
+        )
+
+    @ddt.data(
+        {
+            'version_number': mock.sentinel.version_number,
+            'edition_id': 'server',
+            'installation_type': mock.sentinel.installation_type,
+            'product_name': mock.sentinel.product_name,
+            'distribution_name': windows.WINDOWS_SERVER_IDENTIFIER,
+            'should_raise': False
+        },
+        {
+            'version_number': mock.sentinel.version_number,
+            'edition_id': 'client',
+            'installation_type': mock.sentinel.installation_type,
+            'product_name': mock.sentinel.product_name,
+            'distribution_name': windows.WINDOWS_CLIENT_IDENTIFIER,
+            'should_raise': False
+        },
+        {
+            'should_raise': True
+        }
+    )
+    @mock.patch.object(windows.WindowsOSDetectTools, '_get_image_version_info')
+    def test_detect_os(self, data, mock_get_image_version_info):
+        if data['should_raise']:
```

Comment:
Please have a separate test case when `detect_os` raises exception to avoid this ugly `if` in the body here.